pylint/checkers/strings.py

```diff
@@ -554,15 +562,50 @@ class StringConstantChecker(BaseTokenChecker):
     # Unicode strings.
     UNICODE_ESCAPE_CHARACTERS = 'uUN'
 
+    def __init__(self, *args, **kwargs):
+        super(StringConstantChecker, self).__init__(*args, **kwargs)
+        self.string_tokens = {}  # token position -> (token value, next token)
+
     def process_module(self, module):
         self._unicode_literals = 'unicode_literals' in module.future_imports
 
     def process_tokens(self, tokens):
-        for (tok_type, token, (start_row, _), _, _) in tokens:
+        for i, (tok_type, token, start, _, _) in enumerate(tokens):
             if tok_type == tokenize.STRING:
                 # 'token' is the whole un-parsed token; we can look at the start
                 # of it to see whether it's a raw or unicode string etc.
-                self.process_string_token(token, start_row)
+                self.process_string_token(token, start[0])
+                next_token = tokens[i + 1] if i + 1 < len(tokens) else None
+                try:
+                    self.string_tokens[start] = (ast.literal_eval(token), next_token)
```

Comment:
I'm not super happy about `literal_eval`-ing every string token, as this might result in a performance hit for big projects that rely on multiple strings in their codebase. Can you find a way to avoid doing it?

Reply:
Done